bundles/org.openhab.binding.caddx/src/main/java/org/openhab/binding/caddx/internal/handler/CaddxBridgeHandler.java

```diff
@@ -0,0 +1,408 @@
+                }
+                break;
+            case SEND_COMMAND:
+                if (!command.toString().isEmpty()) {
+                    String[] tokens = command.toString().split("\\|");
+
+                    String cmd = tokens[0];
+                    String data = "";
+                    if (tokens.length > 1) {
+                        data = tokens[1];
+                    }
+
+                    sendCommand(cmd, data);
+
+                    updateState(channelUID, new StringType(""));
+                }
+                break;
+            default:
+                break;
+        }
+    }
+
+    /**
+     * Sends a command to the panel
+     *
+     * @param command The command to be send
+     * @param data The associated command data
+     */
+    public boolean sendCommand(String command, String data) {
+        logger.trace("sendCommand(): Attempting to send Command: command - {} - data: {}", command, data);
+
+        CaddxMessage msg = null;
+
+        if (CaddxBindingConstants.ZONE_BYPASSED.equals(command)) {
+            msg = new CaddxMessage(CaddxMessageType.Zone_Bypass_Toggle, data);
+        } else if (CaddxBindingConstants.ZONE_STATUS_REQUEST.equals(command)) {
+            msg = new CaddxMessage(CaddxMessageType.Zone_Status_Request, data);
+        } else if (CaddxBindingConstants.ZONE_NAME_REQUEST.equals(command)) {
+            msg = new CaddxMessage(CaddxMessageType.Zone_Name_Request, data);
+        } else if (CaddxBindingConstants.PARTITION_STATUS_REQUEST.equals(command)) {
+            msg = new CaddxMessage(CaddxMessageType.Partition_Status_Request, data);
+        } else if (CaddxBindingConstants.PARTITION_PRIMARY_COMMAND.equals(command)) {
+            msg = new CaddxMessage(CaddxMessageType.Primary_Keypad_Function_without_PIN, data);
+        } else if (CaddxBindingConstants.PARTITION_SECONDARY_COMMAND.equals(command)) {
+            msg = new CaddxMessage(CaddxMessageType.Secondary_Keypad_Function, data);
+        } else if (CaddxBindingConstants.PANEL_SYSTEM_STATUS_REQUEST.equals(command)) {
+            msg = new CaddxMessage(CaddxMessageType.System_Status_Request, data);
+        } else if (CaddxBindingConstants.PANEL_INTERFACE_CONFIGURATION_REQUEST.equals(command)) {
+            msg = new CaddxMessage(CaddxMessageType.Interface_Configuration_Request, data);
+        } else if (CaddxBindingConstants.PANEL_LOG_EVENT_REQUEST.equals(command)) {
+            msg = new CaddxMessage(CaddxMessageType.Log_Event_Request, data);
+        } else {
+            logger.trace("Unknown command {}", command);
+            return false;
+        }
+
+        CaddxCommunicator comm = communicator;
+        if (comm != null) {
+            comm.transmit(msg);
+        }
+
+        return true;
+    }
+
+    /**
+     * Register the Discovery Service.
+     *
+     * @param discoveryService
+     */
+    public void registerDiscoveryService(@Nullable CaddxDiscoveryService discoveryService) {
```

Comment:
How would the argument to this method ever be null?

Reply:
You are right. It cannot be null.
(Also code that was copied from the dscalarm binding)
Nullable annotation removed